rust/agama-server/src/software/web.rs

```diff
@@ -153,6 +230,74 @@ async fn products(State(state): State<SoftwareState<'_>>) -> Result<Json<Vec<Pro
     Ok(Json(products))
 }
 
+/// Information about registration configuration (product, patterns, etc.).
+#[derive(Clone, Serialize, Deserialize, utoipa::ToSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct RegistrationInfo {
+    /// Registration key. Empty value mean key not used or not registered.
+    key: String,
+    /// Registration email. Empty value mean email not used or not registered.
+    email: String,
+    /// if registration is required, optional or not needed for current product.
+    /// Change only if selected product is changed.
+    requirement: RegistrationRequirement,
+}
+
+/// returns registration info
+///
+/// * `state`: service state.
+#[utoipa::path(get, path = "/software/registration", responses(
+    (status = 200, description = "registration configuration", body = RegistrationInfo),
+    (status = 400, description = "The D-Bus service could not perform the action")
+))]
+async fn get_registration(
+    State(state): State<SoftwareState<'_>>,
+) -> Result<Json<RegistrationInfo>, Error> {
+    let result = RegistrationInfo {
+        key: state.product.registration_code().await?,
+        email: state.product.email().await?,
+        requirement: state.product.registration_requirement().await?,
+    };
+    Ok(Json(result))
+}
+
+/// Software service configuration (product, patterns, etc.).
+#[derive(Clone, Serialize, Deserialize, utoipa::ToSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct RegistrationParams {
+    /// Registration key.
+    key: String,
+    /// Registration email.
+    email: String,
+}
+
+/// Register product
+///
+/// * `state`: service state.
+#[utoipa::path(post, path = "/software/registration", responses(
```

Comment:
I would most probably use PUT, as you are replacing (or creating) the resource.

Reply:
well, if I use PUT, I will have to create it idempotent, which it is currently not.

Reply:
Well, I am not sure. If you perform the same action with the same data several times, what changes? I would expect the result to be the same. But I am fine with using POST.

Reply:
well, the change will be that after first successful registration the next one fails. Also we do not handle if you use different data. You have to first deregister and then can register with different data. We handle that automatic only when you change product.